lessons/projects/github-api/index.md

```diff
@@ -0,0 +1,178 @@
+# Webové API
+
+Jak už bylo řečeno v [lekci o JSON]({{ lesson_url('intro/json') }}),
+hlavní výhoda formátu JSON je, že se na Internetu rozšířil nejvíc.
+Pojďme toho využít!
+
+Spousta webových služeb poskytuje takzvané
+*API* (z *application programming interface*,
+programátorské rozhraní), přes které je možné s danou
+službou komunikovat programově.
+Místo klikání na tlačítek a čtení stránek „očima”
+dostaneme data ve formátu, kterým rozumí počítače –
+a v dnešní době to bude většinou formát JSON.
+
+Z minulých lekcí bys měl{{a}} mít založený účet na github.com,
+tak se zkusme zeptat Githubu, co o nás ví.
+
+
+## Autorizace
+
+První, a mnohdy nejsložitější, krok k použití API
+je přihlášení. Počítače se totiž přihlašují jinak
+než lidi, a problematika bezpečnosti a oprávnění by vydala na samostatný kurz.
+My to uděláme co nejjednodušeji, ať se rychle dostaneme k jádru věci:
+
+* Přihlaš se na [github.com](https://github.com).
+* Jdi na [nastavení Personal Accesss Tokens](https://github.com/settings/tokens).
+* Vytvoř si nový token ("Generate new token"). Nezaškrtávej žádná oprávnění navíc.
+* Zkopíruj si heslo, které takto dostaneš, do souboru `token.txt`.
+
+!!! warning "Pozor!"
+    Vygenerovaný kód je heslo, které držitele
+    opravňuje pracovat s Githubem pod tvým jménem!
+    Drž ho v tajnosti. Kdyby se přece jen dostalo „ven”, na stránce
+    [Personal Accesss Tokens](https://github.com/settings/tokens) ho deaktivuj.
+
+    
+## Requests
+
+K práci s internetovými stránkami použijeme knihovnu Requests.
+V aktivovaném virtuálním prostředí si ji nainstaluj příkazem:
+
+```console
+(env)$ python -m pip install requests
+```
+
+A potom v Pythonu zkus stáhnout nějakou stránku:
+
+```python
+import requests
+
+# stažení stránky
+stranka = requests.get('https://github.com')
+
+# ověření, že dotaz proběhl v pořádku
+stranka.raise_for_status()
+
+# vypsání obsahu
+print(stranka.text)
+```
+
+Měl by se vypsat obsah stránky
+[https://github.com](https://github.com) –
+HTML kód; který se objeví když v prohlížeči dáš
+„Ukázat zdroj” (*View Page Source*, většinou <kbd>Ctrl</kbd>+<kbd>U</kbd>),
+a ze kterého prohlížeč umí vykreslit stránku.
+
+Ale my nechceme obsah pro lidi.
+Podívejme se, co Github zpřístupňuje počítačům.
+
+
+## Uživatelský účet
+
+Zkus, co dělá tento kód:
+
+```python
+import requests
+
+with open('token.txt') as soubor:
+    token = soubor.read().strip()
+
+headers = {'Authorization': 'token ' + token}
+
+stranka = requests.get('https://api.github.com/user', headers=headers)
+stranka.raise_for_status()
+print(stranka.text)
+```
+
+Co se stalo? Tím, že jsi Githubu dal{{a}} svůj token,
```

Comment:
`{{a}}` is a shortcut for `{{gnd('', 'a'}}`. It's over two times more frequent than other uses of `gnd`.